### PR TITLE
Fix issue #565

### PR DESCRIPTION
**Pull request by jjlauer**
Quick fix for NPE in very small number of routing cases.  Unit tests added.

### CODE REVIEW

File: ninja-core/src/main/java/ninja/Route.java [hunk 1]
```diff
@@ -125,10 +123,12 @@ public boolean matches(String httpMethod, String uri) {
         Matcher m = regex.matcher(uri);
 
         if (m.matches()) {
-            Iterator<String> it = this.parameters.keySet().iterator();
-            for (int i = 1; i < m.groupCount() + 1; i++) {
-                String parameterName = it.next();
-                map.put(parameterName, m.group(i));
+            if (this.parameters != null) {
```
Comment (raphaelbauer) [inline ninja-core/src/main/java/ninja/Route.java]: Another idea. What about returning an empty LinkedList when RouteParameter.parse(uri) instead returning null? Then we don't need any null checks... The test is still cool to make sure stuff works of course...

Not 100% sure but RouteParameter.parse(uri) is only called once when building the Route, so there should not be a performance impact of that...

File: ninja-core/src/site/markdown/team.md [hunk 2]
```diff
@@ -10,10 +10,17 @@ new releases. And driving the platform as a whole.
 That leadership role is currently performed by:
 
  * The team of [FinalFrontierLabs](http://www.finalfrontierlabs.com) 
-   and [Raphael](http://raphaelbauer.com) as lead architect. 
+   and [Raphael](http://raphaelbauer.com) as lead architect.
+
  * The team of [Fizzed, Inc.](http://fizzed.com) and [Joe Lauer](http://lauer.bz)
    as lead architect.
 
+Ninja is proudly sponsored by <a href="https://www.greenback.com">Greenback</a>
```
Comment (raphaelbauer) [inline ninja-core/src/site/markdown/team.md:18]: That's 100% cool - it's great you and your team is part of the project! All the very very best for greenback.com!

**Comment by raphaelbauer** [comment]
Tests don't succeed locally on my machine either (did not do any further debugging...)

**Comment by jjlauer** [comment]
They ran locally for me, but I'll take a look.

**Comment by jjlauer** [comment]
@raphaelbauer Went with your suggestion of just returning an empty map. Travis CI is happy now.

**Comment by raphaelbauer** [comment]
Nice fix!

**Pull request merged**